velox/dwio/parquet/tests/ParquetTpchTest.cpp

```diff
@@ -36,68 +34,93 @@ namespace fs = std::filesystem;
     connector::registerConnector(hiveConnector);
-    tempDirectory_ = exec::test::TempDirectoryPath::create();
-    lineitemType_ =
-        ROW({"orderkey",
-             "partkey",
-             "suppkey",
-             "linenumber",
-             "quantity",
-             "extendedprice",
-             "discount",
-             "tax",
-             "returnflag",
-             "linestatus",
-             "shipdate",
-             "commitdate",
-             "receiptdate",
-             "shipinstruct",
-             "shipmode",
-             "comment"},
-            {BIGINT(),
-             BIGINT(),
-             BIGINT(),
-             BIGINT(),
-             DOUBLE(),
-             DOUBLE(),
-             DOUBLE(),
-             DOUBLE(),
-             VARCHAR(),
-             VARCHAR(),
-             DATE(),
-             DATE(),
-             DATE(),
-             VARCHAR(),
-             VARCHAR(),
-             VARCHAR()});
+
+    // Convert decimal columns to double.
+    const std::string lineitemSQL = R"(
+    COPY (SELECT l_orderkey, l_partkey, l_suppkey, l_linenumber, l_quantity,
+    l_extendedprice::DOUBLE as l_extendedprice,
+    l_discount::DOUBLE as l_discount, l_tax::DOUBLE as l_tax,
+    l_returnflag, l_linestatus, l_shipdate, l_receiptdate,
+    l_shipinstruct, l_shipmode, l_comment
+    FROM {}) TO '{}' (FORMAT 'parquet', ROW_GROUP_SIZE {})
+    )";
+    parquetFilesPaths_["lineitem"] =
+        writeTableToParquet("lineitem", lineitemSQL);
+
+    const std::string ordersSQL = R"(
+    COPY (SELECT o_orderkey, o_custkey, o_orderstatus,
+    o_totalprice::DOUBLE as o_totalprice,
+    o_orderdate, o_orderpriority, o_clerk, o_shippriority, o_comment
+    FROM {}) TO '{}' (FORMAT 'parquet', ROW_GROUP_SIZE {})
+    )";
+    parquetFilesPaths_["orders"] = writeTableToParquet("orders", ordersSQL);
+
+    const std::string customerSQL = R"(
+    COPY (SELECT c_custkey, c_name, c_address, c_nationkey, c_phone,
+    c_acctbal::DOUBLE as c_acctbal, c_mktsegment, c_comment
+    FROM {}) TO '{}' (FORMAT 'parquet', ROW_GROUP_SIZE {})
+    )";
+    parquetFilesPaths_["customer"] =
+        writeTableToParquet("customer", customerSQL);
+
+    planNodeIdGenerator_ = std::make_shared<PlanNodeIdGenerator>();
   }
 
-  void TearDown() override {
+  static void TearDownTestSuite() {
     connector::unregisterConnector(kHiveConnectorId);
     parquet::unregisterParquetReaderFactory();
   }
 
+  /// Write the DuckDB TPC-H table to a Parquet file and return the file path
+  ///
+  /// - tableName is name of table from TPC-H tables
+  /// - parquetWriteSQL is the string containing SQL query to write the
+  ///   tableName as a parquet file to the disk
+  /// - tempDirectory is the location where parquet files are saved
+  ///
+  /// Returns a std::string containing the path to parquet file on disk
+  static std::string writeTableToParquet(
+      const std::string_view& tableName,
+      const std::string_view& parquetWriteSQL) {
```

Comment:
nit: createTableSql or createTableQuery name might be clearer

Reply:
I don't think that matches the action being done inside this static method.  We dumping the table contents of a corresponding TPC-H table into a parquet file. Perhaps `saveTpchTableAsParquetFile` sounds better